src/trimerge-array.ts

```diff
@@ -0,0 +1,89 @@
+import { MergeFn } from './trimerge';
+import { Path } from './path';
+import { JSONObject, JSONValue } from './json';
+import { CannotMerge } from './cannot-merge';
+import { diff3Keys } from './diff3-keys';
+import { jsSameType } from './js-same-type';
+
+type ArrayKeyFn = (item: any, index: number, arrayPath: Path) => string;
+
+export function trimergeArrayCreator(
+  getArrayItemKey: ArrayKeyFn,
+  allowOrderConflicts: boolean = false,
+): MergeFn {
+  return (
+    orig: any[],
+    left: any[],
+    right: any[],
+    path: Path,
+    mergeFn: MergeFn,
+  ): JSONValue[] | typeof CannotMerge => {
+    if (jsSameType(orig, left, right) !== 'array') {
+      return CannotMerge;
+    }
+    const origMap: JSONObject = {};
+    const leftMap: JSONObject = {};
+    const rightMap: JSONObject = {};
```

Comment:
We don't know for sure that `orig`/`left`/`right` are arrays of `JSONObject`s. Should these types be `Record<string, any>`?